### PR TITLE
support specialized csv separator and delimiter

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
Resolve https://github.com/pingcap/dumpling/issues/113

### What is changed and how it works?
Add arguments "csv-separator" and "csv-delimiter" to support specialized csv separator and delimiter.

### Check List 

Tests 

 - Unit test
 - Integration test

Side effects

 - Possible performance regression
 - Increased code complexity

Related changes

 - Need to update the documentation

### CODE REVIEW

File: v4/export/sql_type.go [hunk 1]
```diff
@@ -153,11 +153,11 @@ func (r RowReceiverArr) WriteToBuffer(bf *bytes.Buffer, escapeBackslash bool) {
 	bf.WriteByte(')')
 }
 
-func (r RowReceiverArr) WriteToBufferInCsv(bf *bytes.Buffer, escapeBackslash bool, csvNullValue string) {
+func (r RowReceiverArr) WriteToBufferInCsv(bf *bytes.Buffer, escapeBackslash bool, opt csvOption) {
```
Comment (kennytm) [inline v4/export/sql_type.go:156]: pass this by reference please (also all other occurrences)

```suggestion
func (r RowReceiverArr) WriteToBufferInCsv(bf *bytes.Buffer, escapeBackslash bool, opt *csvOption) {
```

File: v4/export/writer_util_test.go [hunk 2]
```diff
@@ -112,13 +112,34 @@ func (s *testUtilSuite) TestWriteInsertInCsv(c *C) {
...
+
+	bf.Reset()
+	opt.separator = []byte(";")
+	err = WriteInsertInCsv(context.Background(), tableIR, bf, true, opt)
+	c.Assert(err, IsNil)
+	expected = "1;'male';'bob@mail.com';'020-1234';\\N\n" +
+		"2;'female';'sarah@mail.com';'020-1253';'healthy'\n" +
+		"3;'male';'john@mail.com';'020-1256';'healthy'\n" +
+		"4;'female';'sarah@mail.com';'020-1235';'healthy'\n"
+	c.Assert(bf.String(), Equals, expected)
```
Comment (kennytm) [inline v4/export/writer_util_test.go:142]: please include a test case where separator has multiple characters and delimiter is "ma" (`male` should become `mamamalema`, `femail` should become `mafemamalema`, `xxx@mail.com` should become `maxxx@mamail.comma`).

Comment (lichunzhu) [reply v4/export/writer_util_test.go:146]: addressed in https://github.com/pingcap/dumpling/pull/116/commits/7eb97424898876ba047e843605be726c987de83c

File: v4/export/writer_util_test.go [hunk 3]
```diff
@@ -140,6 +144,20 @@ func (s *testUtilSuite) TestWriteInsertInCsv(c *C) {
 		"3;'male';'john@mail.com';'020-1256';'healthy'\n" +
 		"4;'female';'sarah@mail.com';'020-1235';'healthy'\n"
 	c.Assert(bf.String(), Equals, expected)
+
+	// test delimiter that included in values
+	bf.Reset()
+	opt.separator = []byte(",")
```
Comment (kennytm) [inline v4/export/writer_util_test.go:150]: Separator should be multi-byte too

Comment (lichunzhu) [reply v4/export/writer_util_test.go]: addressed in https://github.com/pingcap/dumpling/pull/116/commits/bd8134310936180d35dfeedb2ac5d1b78c9446c1

**Pull request merged**